sympy/matrices/expressions/tests/test_derivatives.py

```diff
@@ -27,7 +27,7 @@
 from sympy.matrices.expressions import hadamard_power
 from sympy.tensor.array.expressions.array_expressions import ArrayAdd, ArrayTensorProduct, PermuteDims
 
-k = symbols("k")
+k, = symbols("k,")
```

Comment:
Does this imply the former statement no longer works? if so, I don't think this test should be changed.

Reply:
No. The former statement still works at runtime. It just mean mypy complains about it.

Reply:
I'm not sure about this. We need to be clear about whether using symbols like this is discouraged because otherwise mypy is going to complain about user code.

Reply:
I don't think we should have to write unusual forms of code like this to satisfy a type checker. Seems like the type checker is wrong in this case.

Reply:
> I'm not sure about this. We need to be clear about whether using symbols like this is discouraged because otherwise mypy is going to complain about user code.

If a user uses mypy, it means the user definitely wants stronger type support and inevitably fix code according to type checker. Original `symbols` provides nothing.

Reply:
I tried editing a file in vscode with pylance (which is the "recommended" plugin). If I write `x = symbols('x')` and then hover the mouse over `x` it says that `x` has type `Tuple[Symbol, ...]` which is incorrect.

Reply:
> I tried editing a file in vscode with pylance (which is the "recommended" plugin). If I write `x = symbols('x')` and then hover the mouse over `x` it says that `x` has type `Tuple[Symbol, ...]` which is incorrect.

True. That's why we'd better not suggest writing `symbols('x')`. Is there a scene where it cannot be replaced by `Symbol('x')`?

Reply:
Why can't we make symbols return a single element as well?

Reply:
I think the problem is that a correct type hint here would look something like
```python
def symbols(...) -> Symbol | Tuple[Symbol]
```
but then when you do
```python
x = symbols('x')
```
the type checker would not be able to infer that x has type `Symbol`. Then something like `x.is_positive` would appear to be a type failure (tuple has no attribute `is_positive`). The problem is that `symbols('x')` and `symbols('x, y')` return different types but the type is determined by the *value* of the argument rather than the types of the arguments.